### ISSUE TITLE
Precision notes

### ISSUE THREAD

**Issue by alex-hh** [alex-hh/profam#363]
Lightning distinguishes between mixed and true setttings:
in mixed, model weights are stored in fp32. in true weights are stored in specified precision.

https://lightning.ai/docs/fabric/stable/fundamentals/precision.html#true-half-precision
> As mentioned before, for numerical stability mixed precision keeps the model weights in full float32 precision while casting only supported operations to lower bit precision. However, in some cases it is indeed possible to train completely in half precision. Similarly, for inference the model weights can often be cast to half precision without a loss in accuracy (even when trained with mixed precision).

16-bit Precision
Use 16-bit mixed precision to speed up training and inference. If your GPUs are [[Tensor Core](https://docs.nvidia.com/deeplearning/performance/mixed-precision-training/index.html)] GPUs, you can expect a ~3x speed improvement.

Trainer(precision="16-mixed")
In most cases, mixed precision uses FP16. Supported [PyTorch operations](https://pytorch.org/docs/2.4/amp.html#op-specific-behavior) automatically run in FP16, saving memory and improving throughput on the supported accelerators. Since computation happens in FP16, which has a very limited “dynamic range”, there is a chance of numerical instability during training. This is handled internally by a dynamic grad scaler which skips invalid steps and adjusts the scaler to ensure subsequent steps fall within a finite range. For more information [see the autocast docs](https://pytorch.org/docs/2.4/amp.html#gradient-scaling).

With true 16-bit precision you can additionally lower your memory consumption by up to half so that you can train and deploy larger models. However, this setting can sometimes lead to unstable training.

Trainer(precision="16-true")
